plugins/outputs/azure_data_explorer/azure_data_explorer_test.go

```diff
@@ -193,6 +196,111 @@ func TestWrite(t *testing.T) {
 	}
 }
 
+/***/
+
+func TestWriteWithType(t *testing.T) {
+	metricName := "test1"
+	fakeClient := kusto.NewMockClient()
+	expectedResultMap := map[string]string{metricName: `{"fields":{"value":1},"name":"test1","tags":{"tag1":"value1"},"timestamp":1257894000}`}
+	mockMetrics := testutil.MockMetrics()
+	// Multi tables
+	mockMetrics2 := testutil.TestMetric(1.0, "test2")
+	mockMetrics3 := testutil.TestMetric(2.0, "test3")
+	mockMetricsMulti := make([]telegraf.Metric, 2)
+	mockMetricsMulti[0] = mockMetrics2
+	mockMetricsMulti[1] = mockMetrics3
+	expectedResultMap2 := map[string]string{"test2": `{"fields":{"value":1.0},"name":"test2","tags":{"tag1":"value1"},"timestamp":1257894000}`, "test3": `{"fields":{"value":2.0},"name":"test3","tags":{"tag1":"value1"},"timestamp":1257894000}`}
+	// List of tests
+	testCases := []struct {
+		name                      string
+		inputMetric               []telegraf.Metric
+		metricsGrouping           string
+		tableNameToExpectedResult map[string]string
+		expectedWriteError        string
+		createTables              bool
+		ingestionType             string
+	}{
+		{
+			name:                      "Valid metric",
+			inputMetric:               mockMetrics,
+			createTables:              true,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap,
+		},
+		{
+			name:                      "Don't create tables'",
+			inputMetric:               mockMetrics,
+			createTables:              false,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap,
+		},
+		{
+			name:                      "SingleTable metric grouping type",
+			inputMetric:               mockMetrics,
+			createTables:              true,
+			metricsGrouping:           singleTable,
+			tableNameToExpectedResult: expectedResultMap,
+		},
+		{
+			name:                      "Valid metric managed ingestion",
+			inputMetric:               mockMetrics,
+			createTables:              true,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap,
+			ingestionType:             managedIngestion,
+		},
+		{
+			name:                      "Table per metric type",
+			inputMetric:               mockMetricsMulti,
+			createTables:              true,
+			metricsGrouping:           tablePerMetric,
+			tableNameToExpectedResult: expectedResultMap2,
+		},
+	}
+	for _, testCase := range testCases {
+		testCase := testCase
```

Comment:
The test cases are made to run in parallel, so this is done to avoid issues that come with asynchronous behavior. 

ref: [Be Careful with Table Driven Tests and t.Parallel()](https://gist.github.com/posener/92a55c4cd441fc5e5e85f27bca008721#be-careful-with-table-driven-tests-and-tparallel)

Reply:
Then don't run them in parallel. :-) I don't expect a big benefit, am I mislead?